### DISCUSSION TITLE
Using the Imaginate feature

### DISCUSSION THREAD

**Discussion by Njasa2k** [GraphiteEditor/Graphite#1056] [Help] [answered]
I was wondering if the imaginate function of the editor works, is it a mockup for future functionality or am I using it wrong?

**Comment by Keavon** [comment, marked as answer]
It does work but we unfortunately have no documentation yet. You have to self-host it using an installation of the [A1111 Stable Diffusion web UI](https://github.com/AUTOMATIC1111/stable-diffusion-webui), launched with the `--api --cors-allow-origins=*` flags. I believe there may be a (very slow) CPU fallback and maybe an AMD GPU-compatible version that might work (I haven't tested it), but please be aware this normally this requires an Nvidia GPU with at least 6 GB of VRAM. [This issue](https://github.com/GraphiteEditor/Graphite/issues/1002) will be worked on pretty soon, which will allow a crowd-sourced network of users to provide the hosting of Imaginate for users (but lacking some features, like previewing the generation progress). I also plan to personally host the fully-featured A1111 server on my own machine as a perk for those who sponsor the project, but I haven't sorted out those details yet. I'm aiming to get this all done, optimistically, in the next week.

**Reply by Njasa2k** [reply]
Thank you, I am currently using it and it seems to works within 3 seconds.

**Reply by Keavon** [reply]
Fantastic! Is that with Nvidia, AMD, or CPU?

**Reply by Njasa2k** [reply]
Nvidia RTX 2070

**Reply by Keavon** [reply]
Perfect, I have the same card (the Super version) and it works nicely indeed.

**Comment by Keavon** [comment]
Further information is here: https://github.com/GraphiteEditor/Graphite/discussions/1089
